SequenceReplay/SequenceReplay.py

```diff
@@ -255,95 +256,103 @@ def setParameterNode(self, inputParameterNode: Optional[SequenceReplayParameterN
         sequenceBrowsersLayout = self.ui.sequenceBrowsersCollapsibleButton.layout()
-        
-        # Remove all buttons from the layout
-        for i in reversed(range(sequenceBrowsersLayout.count())):
-            widget = sequenceBrowsersLayout.itemAt(i).widget()
-            widget.setParent(None)
-            widget.deleteLater()
-            
-        # Create a button for each node
+        self._clearLayout(sequenceBrowsersLayout)
+
+        sequenceBrowserNodes = slicer.util.getNodesByClass("vtkMRMLSequenceBrowserNode")
         for node in sequenceBrowserNodes:
-            button = qt.QPushButton(node.GetName())
+            button = qt.QPushButton("Play " + node.GetName())
             button.setCheckable(False)
-            button.clicked.connect(lambda checked, buttonId=node.GetID(): self.onSequenceBrowserButtonClicked(buttonId))
+            button.clicked.connect(lambda checked, nodeId=node.GetID(): self.onSequenceBrowserButtonClicked(nodeId))
             sequenceBrowsersLayout.addWidget(button)
-        
-        # Find all transformable nodes in the scene and create a checkbox for each of them
-        
+            self.buttonState[node.GetID()] = button  # Store button in dictionary
+
+    def _updateTransformableNodeCheckboxes(self):
         nodeTypes = ["vtkMRMLTransformNode", "vtkMRMLVolumeNode", "vtkMRMLMarkupsNode"]
-        
-        transformableNodeIds = []
-        
-        for nodeType in nodeTypes:
-            nodeList = slicer.util.getNodesByClass(nodeType)
-            for node in nodeList:
-                if not node.GetHideFromEditors():
-                    transformableNodeIds.append(node.GetID())
-                    
-        # Check if transformableNodeIds is different from currentTransformableNodeIds
+        transformableNodeIds = [
+            node.GetID() for nodeType in nodeTypes for node in slicer.util.getNodesByClass(nodeType) if
+            not node.GetHideFromEditors()
+        ]
+
         if transformableNodeIds != self.currentTransformableNodeIds:
             self.currentTransformableNodeIds = transformableNodeIds
-            
             checkboxesLayout = self.ui.nodesCollapsibleButton.layout()
-            
-            # Remove all checkboxes from the layout
-            for i in reversed(range(checkboxesLayout.count())): 
-                checkboxesLayout.itemAt(i).widget().setParent(None)
-                checkboxesLayout.itemAt(i).widget().deleteLater()
-            
-            # Create a checkbox for each node
+            self._clearLayout(checkboxesLayout)
+
             for nodeId in transformableNodeIds:
                 node = slicer.mrmlScene.GetNodeByID(nodeId)
-                checkBox = qt.QCheckBox(node.GetName())
-                checkboxesLayout.addWidget(checkBox)
+                if node:
+                    checkBox = qt.QCheckBox(node.GetName())
+                    checkboxesLayout.addWidget(checkBox)
+
+    def _clearLayout(self, layout):
```

Comment:
you can make it a staticmethod to indicate that it is just some global utility function and does not use or modify `self`.